package/kernel/r8168/patches/0002-print-link-speed-and-duplex-mode.patch

```diff
@@ -0,0 +1,39 @@
+From 1e7b4b74a61c6627e1051ef67f11fa11e2c365dc Mon Sep 17 00:00:00 2001
+From: Chukun Pan <amadeus@jmu.edu.cn>
+Date: Sun, 4 Aug 2024 16:15:12 +0800
+Subject: [PATCH] r8168: print link speed and duplex mode
+
+Like other Ethernet drivers, print link speed and duplex mode
+when the interface is up. Formatting output at the same time.
+
+Signed-off-by: Chukun Pan <amadeus@jmu.edu.cn>
+---
+ src/r8168_n.c | 8 ++++++--
+ 1 file changed, 6 insertions(+), 2 deletions(-)
+
+--- a/src/r8168_n.c
++++ b/src/r8168_n.c
+@@ -5377,6 +5377,7 @@ static void
+ rtl8168_check_link_status(struct net_device *dev)
+ {
+         struct rtl8168_private *tp = netdev_priv(dev);
++        u16 status = RTL_R8(tp, PHYstatus);
```

Comment:
AFAIK this could be moved under if (netif_msg_ifup(tp)) because we're only using it there.

Reply:
BTW, this should be `u8` (RTL_R8).